src/bom/extract.rs

```diff
@@ -85,29 +85,19 @@ pub fn do_bitcode_extraction(extract_options : &ExtractOptions,
     // NOTE: Ideally, we would be able to use the tar library for this instead
     // of calling out to tar.
 
-    extract_ops.push_op(
-        &SubProcOperation::new(
-            &Executable::new("tar",
-                             ExeFileSpec::Append,
-                             ExeFileSpec::NoFileUsed))
-            .set_label("tar:unpack")
-            .push_arg("xif")
-            .set_dir(tmp_path)  // Extracted files will be placed here
-            // Input here is explicitly the section dump target file rather
-            // than the output of the previous op.
-            .set_input_file(&FileArg::loc(tar_path))
-    );
+    extract_ops.push_op(&run(&TAR_EXTRACT, &None)
+                        .set_label("tar:unpack")
+                        .set_dir(tmp_path)  // Extracted files will be placed here
+                        // Input here is explicitly the section dump target file
+                        // rather than the output of the previous op.
+                        .set_input_file(&FileArg::loc(tar_path)));
 
     // Now all the files extracted from bitcode.tar into tmp_dir should be linked
     // together to create the final bitcode file.
 
-    let linker = String::from(extract_options.llvm_link_path
-                              .as_ref().unwrap_or(&"llvm-link"
-                                                  .to_string()));
     let mut link_bc_files = extract_ops.push_op(
-        &SubProcOperation::new(&Executable::new(&linker,
-                                                ExeFileSpec::Append,
-                                                ExeFileSpec::option("-o"))));
+        &run(&*LLVM_LINK, &extract_options.llvm_link_path));
```

Comment:
I'm a bit surprised to see the `&*` - is that needed? Related, I was a bit surprised to see `LLVM_LINK` and friends declared as refs

Reply:
`lazy_static` requires `ref` because it performs the actual initialization on the deref (the `*`), but the `run` takes a reference (the `&`), so the `&*` is required to deref the static and then generate a reference for the result.